java/impl/aws/src/main/java/co/worklytics/psoxy/aws/request/APIGatewayV1ProxyEventRequestAdapter.java

```diff
@@ -0,0 +1,92 @@
+package co.worklytics.psoxy.aws.request;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.google.common.base.Splitter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor(staticName = "of")
+public class APIGatewayV1ProxyEventRequestAdapter implements co.worklytics.psoxy.gateway.HttpEventRequest {
+
+
+    final @NonNull APIGatewayProxyRequestEvent event;
+
+    private Map<String, String> caseInsensitiveHeaders;
+
+    @Override
+    public String getPath() {
+        return StringUtils.prependIfMissing(event.getPath(), "/");
+    }
+
+    @Override
+    public Optional<String> getQuery() {
+        if (event.getQueryStringParameters() == null || event.getQueryStringParameters().isEmpty()) {
+            return Optional.empty();
+        } else {
+            return Optional.of(event.getQueryStringParameters().entrySet().stream()
+                .map(e -> e.getKey() + "=" + e.getValue())
+                .collect(Collectors.joining("&")));
+        }
+    }
+
+    @Override
+    public Optional<String> getHeader(String headerName) {
+        return Optional.ofNullable(getCaseInsensitiveHeaders().get(headerName.toLowerCase()));
+    }
+
+    @Override
+    public Optional<List<String>> getMultiValueHeader(String headerName) {
+        return getHeader(headerName.toLowerCase()).map(s -> Splitter.on(',').splitToList(s));
+    }
+
+    @Override
+    public String getHttpMethod() {
+        if (event.getRequestContext().getHttpMethod() == null) {
+            //q: better exception to throw here???
+            throw new IllegalStateException("Psoxy expects API Gateway V1 REST API proxy payload here. If using API Gateway V2 or Lambda function, please set the handler as co.worklytics.psoxy.Handler");
```

Comment:
I think the text is fine; however not sure if worth to return a 502 too (bad gateway? not sure if very accurate, as we might to represent that psoxy config is wrong)

Reply:
I think any uncaught exception (which is what this will end up as) will return a 502 if a lambda url, and maybe a 500 through API gateway - from my experience.

I think just relying on that behavior is fine for now; people need to grab the logs if they see any error case.